task_test.go

```diff
@@ -24,12 +25,20 @@ import (
 
 func TestAllTasks(t *testing.T) {
 	client := NewFakeMarathonEndpoint(t)
-	tasks, err := client.AllTasks()
+	tasks, err := client.AllTasks(nil)
 	assert.Nil(t, err)
 	assert.NotNil(t, tasks)
 	assert.Equal(t, len(tasks.Tasks), 2)
 }
 
+func TestAllStagingTasks(t *testing.T) {
+	client := NewFakeMarathonEndpoint(t)
+	tasks, err := client.AllTasks(url.Values{"status": []string{"staging"}})
```

Comment:
> I think I would rather help a user by explicitly providing possible parameters. In this case you don't need to look to the api and you have compile time check that your params are right.
There are two downsides with this approach:
1) as it was mentioned: api might change
2) providing nil values will be ugly
We can solve 2) by wrapping request params into structs, and I think it might look actually nice.
I think at this point I will do url.Values and then we can decide if we want to refactor the way parameters are passed.

So my suggestion would be something like:
```
client.AllTasks(AllTasksRequest{status: marathon.TaskStaging})
```

Or since marathon doesn't have more then 2 parameters the first version even could be ok:
```
client.AllTasks(marathon.TaskAny)
// or 
client.AllTasks(maraton.TaskStaging)
```
Again, it's just my preference :)